### PR TITLE
feat(Listbox): auto dense

### PR DESCRIPTION
**Pull request by DanielS-Qlik**
## Motivation

When the listbox container is smaller than a grid item, enable dense mode.
fixes https://github.com/qlik-oss/sn-list-objects/issues/214 together with https://github.com/qlik-oss/sn-list-objects/pull/220

Kapture 2023-03-28 at 14 43 32

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [x] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

**Comment by Caele** [comment]
There is a lot of logic and properties being passed around here, quite hard to follow its exact purpose.

What is is that triggers the auto dense setting? And does it simply result in dense being true?

**Comment by DanielS-Qlik** [comment]
> There is a lot of logic and properties being passed around here, quite hard to follow its exact purpose.
> 
> What is is that triggers the auto dense setting? And does it simply result in dense being true?

Basically it checks the height of element passed in here https://github.com/qlik-oss/nebula.js/blob/bb82663c84116a6293f0e9621eabfbe5218b0267/apis/nucleus/src/index.js#L458.
If it's in grid mode and that element is smaller than a grid item, it enables dense mode via `autoDense`.

The state needs to be kept in ListboxInline.jsx because it's used by the `ListBoxSearch` there.
Then the state is set in ListBox.jsx via `useSetAutoDense()` because it needs the loaderRef to detect the scollbar presence.

**Comment by veinfors** [comment]
I think we can wait with merging this as we might have better ideas to improve the behaviour instead...
(introducing a setting to the sn-filter-pane to explicitly control wether listboxes should collapse or not could help)
